### PR TITLE
virt-test: utils_test.qemu: Add some func() for windows driver verifier.

### PR DESCRIPTION
**Pull request by CongSmile**
1. Enable the windows driver verifier.
2. Verify the windows driver verifier.
3. Clear the windows driver verifier.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1256248

### CODE REVIEW

**Comment by CongSmile** [comment]
Hi @vipmike007

Could you help review it?

Thanks.

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2451,6 +2451,55 @@ def get_windows_drive_letters(session):
...
+    :param vm: VM object.
+    :param timeout: Timeout in seconds.
+    :return: Session after reboot.
+    """
+    verifier_setup_cmd = "verifier.exe /standard /driver %s" % driver
+    if verifier_setup_cmd:
+        session.cmd(verifier_setup_cmd,
+                    timeout=timeout,
+                    ignore_all_errors=True)
+    return vm.reboot(session)
```
Comment (vipmike007) [inline virttest/utils_misc.py]: why not reboot it in the test ?

Comment (CongSmile) [inline virttest/utils_misc.py]: In my opinion, rebooting is a must part for setup_win_driver_verifier() to enable the verifier.
it's better to add it in setup_win_driver_verifier().

Thanks.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -2451,6 +2451,55 @@ def get_windows_drive_letters(session):
     return drive_letters
 
 
+def setup_win_driver_verifier(session, driver, vm, timeout):
+    """
```
Comment (vipmike007) [inline virttest/utils_misc.py]: Do we need to set a default timeout value ?

Comment (CongSmile) [inline virttest/utils_misc.py]: Ok, that's fine.

Thanks for the review, will update it.

Comment (CongSmile) [inline virttest/utils_misc.py]: Hi @vipmike007 

I have added the default timeout.
Could you help review it again?

Thanks.

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -2451,6 +2451,55 @@ def get_windows_drive_letters(session):
...
+    :param timeout: Timeout in seconds.
+    """
+    verifier_query_cmd = "verifier.exe /querysettings"
+    output = session.cmd_output(verifier_query_cmd, timeout=timeout)
+    if not re.search(r"%s" % driver, output, re.M):
+        msg = "Verify device driver failed, "
+        msg += "guest report driver is %s, " % output
+        msg += "expect is '%s'" % driver
+        raise error.TestFail(msg)
+
```
Comment (vipmike007) [inline virttest/utils_misc.py]: Failed to review:

testing code 
utils_misc.setup_win_driver_verifier(session, driver_name, vm,iozone_timeout)
utils_misc.verify_win_driver_verifier(session,driver_name, iozone_timeout)

 Exception escaping from test:
Traceback (most recent call last):
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 411, in _exec
    _call_test_function(self.execute, *p_args, **p_dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 830, in _call_test_function
    raise error.UnhandledTestFail(e)
UnhandledTestFail: Unhandled ShellProcessTerminatedError: Shell process terminated while waiting for command to complete: 'verifier.exe /querysettings'    (status: 1,    output: '')    [context: Generating qemu-kvm commandline]
Traceback (most recent call last):
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 823, in _call_test_function
    return func(*args, **dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 291, in execute
    postprocess_profiled_run, args, dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 209, in _call_run_once
    *args, **dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 313, in run_once_profiling
    self.run_once(*args, **dargs)
  File "/root/autotest/test/autotest-devel/client/tests/virt/virt.py", line 155, in run_once
    run_func(self, params, env)
  File "/root/autotest/test/autotest-devel/client/shared/error.py", line 141, in new_fn
    return fn(*args, **kwargs)
  File "/root/autotest/test/autotest-devel/client/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/windows_disk_smoke_test.py", line 76, in run
    utils_misc.verify_win_driver_verifier(session,driver_name, iozone_timeout)
  File "/root/autotest/test/autotest-devel/client/tests/virt/virttest/utils_misc.py", line 2481, in verify_win_driver_verifier
    output = session.cmd_output(verifier_query_cmd, timeout=timeout)
  File "/root/autotest/test/autotest-devel/client/tests/virt/virttest/aexpect.py", line 1475, in cmd_output
    raise ShellProcessTerminatedError(cmd, e.status, o)
ShellProcessTerminatedError: Shell process terminated while waiting for command to complete: 'verifier.exe /querysettings'    (status: 1,    output: '')

Comment (CongSmile) [inline virttest/utils_misc.py]: there should get the new session:
session = utils_misc.setup_win_driver_verifier(session, driver_name, vm,iozone_timeout)
utils_misc.verify_win_driver_verifier(session,driver_name, iozone_timeout)

Comment (vipmike007) [inline virttest/utils_misc.py]: Yes, You are right.

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -2451,6 +2451,55 @@ def get_windows_drive_letters(session):
...
+    :param session: VM session.
+    :param timeout: Timeout in seconds.
+    """
+    verifier_clear_cmd = "verifier.exe /reset"
+    if verifier_clear_cmd:
+        session.cmd(verifier_clear_cmd,
+                    timeout=timeout,
+                    ignore_all_errors=True)
+
```
Comment (vipmike007) [inline virttest/utils_misc.py]: pls reboot guest to make sure reset is take affect.

**Comment by vipmike007** [comment]
LGTM.

Ack'ed-By: Mike Cao /<bcao@redhat.com/>

**Comment by CongSmile** [comment]
Hi @xutian 

Yes, you are right, thanks.

I have updated the code, could you help review it again?

Thanks.

File: virttest/utils_misc.py [hunk 5]
```diff
@@ -2451,6 +2451,48 @@ def get_windows_drive_letters(session):
...
+    :param timeout: Timeout in seconds.
+    :return: Session after reboot.
+    """
+    verifier_setup_cmd = "verifier.exe /standard /driver %s.sys" % driver
+    if driver:
+        session.cmd(verifier_setup_cmd,
+                    timeout=timeout,
+                    ignore_all_errors=True)
+    session = vm.reboot(session)
+
```
Comment (xutian) [inline virttest/utils_misc.py]: why blank line here ?

**Comment by xutian** [comment]
This function should a parts of test steps,  so I think it better to move it into utils_test module.

Thanks,
Xu

**Comment by xutian** [comment]
And it should be print steps info in this function like: error.context("enable verifyer in guest OS") , error.context("disable verifyer in guest OS")

File: virttest/utils_misc.py (hunk 5 above)
Comment (CongSmile) [inline virttest/utils_misc.py]: Just want to make the func more clear.

The part before blank line is enable verifier, the others are verify it.

**Comment by CongSmile** [comment]
As I think it's better to add such context in the case it better.

**Comment by xutian** [comment]
OK,  Please move your code to uitls_test.qemu module.

Thanks,
Xu

**Comment by xutian** [comment]
Most of function in this module is host operation,  utils_test module is  right place  to keep test step like functions.

Thanks,
Xu

**Comment by CongSmile** [comment]
Ok, have updated it.

Could you help review it again?

Thanks.

**Comment by xutian** [comment]
LGTM,

Merged, thanks @CongSmile  and @vipmike007

**Pull request merged**